pkgs/tools/misc/btrfs-assistant/default.nix

```
@@ -0,0 +1,91 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+, bash
+, btrfs-progs
+, btrfs-snap
+, cmake
+, coreutils
+, git
+, pkg-config
+, qtbase
+, qtsvg
+, qttools
+, snapper
+, util-linux
+, wrapQtAppsHook
+, enableSnapper ? false
+}:
+
+let
+  runtimeDeps = lib.makeBinPath ([
+    coreutils
+    snapper
+    util-linux
+  ]
+  ++ lib.optional enableSnapper snapper);
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "btrfs-assistant";
+  version = "1.8";
+
+  src = fetchFromGitLab {
+    owner = "btrfs-assistant";
+    repo = "btrfs-assistant";
+    rev = "${finalAttrs.version}";
+    hash = "sha256-Ay2wxDVue+tG09RgAo4Zg2ktlq6dk7GdIwAlbuVULB4=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    git
+    pkg-config
+  ];
+
+  buildInputs = [
+    btrfs-progs
+    btrfs-snap
+    qtbase
+    qtsvg
+    qttools
+  ];
+
+  propagatedBuildInputs = [ wrapQtAppsHook ];
+
+  prePatch = ''
+    substituteInPlace src/util/System.cpp \
+      --replace '/bin/bash' "${bash}/bin/bash"
+  ''
+  + lib.optionalString enableSnapper ''
+    substituteInPlace src/main.cpp \
+      --replace '/usr/bin/snapper' "${snapper}/bin/snapper"
+  '';
+
+  postPatch = ''
+    substituteInPlace src/org.btrfs-assistant.pkexec.policy \
+      --replace '/usr/bin' "$out/bin"
+
+    substituteInPlace src/btrfs-assistant \
+      --replace 'btrfs-assistant-bin' "$out/bin/btrfs-assistant-bin"
+
+    substituteInPlace src/btrfs-assistant-launcher \
+      --replace 'btrfs-assistant' "$out/bin/btrfs-assistant"
+  ''
+  + lib.optionalString enableSnapper ''
+    substituteInPlace src/btrfs-assistant.conf \
+      --replace '/usr/bin/snapper' "${snapper}/bin/snapper"
+  '';
+
+  qtWrapperArgs = [
+    "--prefix PATH : ${runtimeDeps}"
+  ];
```

Usually we "factor" a variable when it is used in more than one place. When it is used in only one place like that, we bring the `let . . . in` right near:

```suggestion
  qtWrapperArgs = let
    runtimeDeps = lib.makeBinPath ([
      coreutils
      util-linux
    ]
    ++ lib.optional enableSnapper snapper);
 in [
   "--prefix PATH : ${runtimeDeps}"
 ];
```